cmd/create.go

```diff
@@ -429,6 +434,45 @@ func (cc *createCmd) createFiles(detectedLang *config.DraftConfig, lowerLang str
 	return nil
 }
 
+func (cc *createCmd) detectDefaults(detectedLang *config.DraftConfig, lowerLang string) {
+	if lowerLang == "gradlew" || lowerLang == "gradle" {
+		// read from build.gradle and detect version
+		f, err := os.Open("build.gradle")
+		if err != nil {
+			log.Warn("Unable to read build.gradle, skipping detection")
+			return
+		}
+		defer f.Close()
+		detectedDefaults := make([]config.BuilderVarDefault, 0)
+		scanner := bufio.NewScanner(f)
+		for scanner.Scan() {
+			line := scanner.Text()
+			if strings.Contains(line, "sourceCompatibility") {
```

Comment:
I'm not familiar with Gradle at all but just curious about syntax requirements.

is "sourceCompatibility" always in that casing or can users use "SourceCompatibility"?

Same for the below things

is the version always separated by exactly one space? can they use other whitespace characters other than space (like tab)? 

Similar questions with the quotes. Is the version always surrounded by single tick quotes or can it be "1.8"?

Reply:
from a quick search it seems like the quotes at least can be "

https://www.baeldung.com/gradle-sourcecompatiblity-vs-targetcompatibility#handling-versions-with-gradle

Reply:
@OliverMKing I can confirm you on these soon. Thank you.

Reply:
yes, casing is important, here is the link for it in official gradle docs. https://docs.gradle.org/current/userguide/building_java_projects.html#using_java_compatibility_options

![image](https://github.com/Azure/draft/assets/105889062/cabe014b-fdb7-4368-9a10-ef50b81c6809)

Although there are other options have come recently with latest gradle ex: java tooling, release property which I will enhance in this PR further upon reading much about it. I have used personally only source and target compatibility. 

Also good call out on single vs double quotes both are possible, also with and without spaces. usually build.gradle will have spaces around quotes but will account them for without spaces as well. Thanks.

Reply:
Updated to handle with and without spaces, single and double quotes.